tensorflow/contrib/avro/python/parse_avro_record.py

```diff
@@ -0,0 +1,476 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import os
+import re
+
+from tensorflow.python.eager import context as _context
+from tensorflow.python.eager import execute as _execute
+
+from tensorflow.python.framework import constant_op
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import ops
+from tensorflow.python.framework import sparse_tensor
+from tensorflow.python.framework import tensor_shape
+
+from tensorflow.python.framework import load_library
+from tensorflow.core.framework import attr_value_pb2 as _attr_value_pb2
+from tensorflow.core.framework import op_def_pb2 as _op_def_pb2
+from tensorflow.core.framework import types_pb2 as _types_pb2
+
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops.parsing_ops import _features_to_raw_params, _prepend_none_dimension, VarLenFeature, \
+    SparseFeature, FixedLenFeature, FixedLenSequenceFeature, _construct_sparse_tensors_for_sparse_features
+
+# HACK: Change to tensorflow.contrib.avro when tensorflow/contrib/__init__.py
+from tensorflow.contrib.avro.python.avro_record_dataset import _create_op_def_library
+
+# Load the shared library
+this_dir = os.path.dirname(os.path.abspath(__file__))
+lib_name = os.path.join(this_dir, '_parse_avro_record.so')  # Display the operators with lib_name.OP_LIST
+parse_module = load_library.load_op_library(lib_name)  # Load the library with dependent so's in '.'
+
+
+def parse_avro_record(serialized, schema, features):
+    """
+    Parses serialized avro records into TensorFlow tensors. This method also handles a batch of serialized records and
```

Comment:
This also doesn't conform to Google style (long lines here), but unfortunately I don't know of a way to automatically fix everything... here are instructions for running pylint:

https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md#python-coding-style

Reply:
Thanks for noting this. Our current code got a score -3.4 out of 10 (mostly common formatting problems: wrong indentation, wrong import order, lines too long).

Can we postpone this until later? The problem: The default style check we use differs and I need to change the setup--aside from the tedious reformatting.

Reply:
We did format the code.